include/ozo/connection.h

```diff
@@ -319,6 +319,16 @@ struct async_get_connection_impl {
     }
 };
 
+template <typename, typename = std::void_t<>>
+struct is_connection_source : std::false_type {};
+template <typename T>
+struct is_connection_source<T, std::void_t<
```

Comment:
IMO it is better to make stronger concept check like for **ConnectionProvider**, something like this
```C++
template <typename T>
struct is_connection_source<T, std::void_t<decltype(
    std::declval<T&>()(
        std::declval<io_context&>(), 
        std::declval<std::function<void(error_code, connection_type<T>)>>()
    )
)>> : std::true_type {};
```